cmp/clock/Clock.js

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2019 Extremely Heavy Industries Inc.
+ */
+
+import {box, span} from '@xh/hoist/cmp/layout';
+import {hoistCmp, HoistModel, managed, useLocalModel, XH} from '@xh/hoist/core';
+import {action, observable} from '@xh/hoist/mobx';
+import {MINUTES, ONE_SECOND} from '@xh/hoist/utils/datetime';
+import {Timer} from '@xh/hoist/utils/async';
+import {isEmpty, isNil, isNumber} from 'lodash';
+import moment from 'moment';
+import PT from 'prop-types';
+
+/**
+ * A component to display the current time.
+ *
+ * Automatically updates on a regular interval to stay current.
+ */
+export const [Clock, clock] = hoistCmp.withFactory({
+    displayName: 'Clock',
+    className: 'xh-clock',
+
+    render({timezone, format, updateInterval, prefix, suffix, ...props}) {
+        const impl = useLocalModel(LocalModel);
+        impl.setData({timezone, format, updateInterval, prefix, suffix});
```

Comment:
I see that we're using this convention of having a `setData()` on a local model we configure from props - or at least I see it also exists in `relativeTimestamp`.  What about `configure()` instead, or something more like that?

Reply:
I think this is a pattern we are going to see a lot.  Could be something like "noteRenderedProps" or "setLatestProps".

Something about the fact that this is called on every render. 

 We could even make this -- *gulp* -- a new optional built-in lifecycle method, if useLocalModel was provided with props....   (not sure about that -- but something worth thinking about ).